Sources/VSM/Documentation.docc/ComprehensiveGuide/ModelDefinition.md

```diff
@@ -184,134 +183,134 @@ First, we'll take a look at the shape of the `EditUserProfileViewState` and its
 struct EditUserProfileViewState {
     var data: UserData
     var editingState: EditingState
-    
+
     enum EditingState {
         case editing(EditingModel)
         case saving
         case savingError(ErrorModel)
     }
-    
-    struct EditingModel {
-        let saveUsername: (String) -> AnyPublisher<EditUserProfileViewState, Never>
-    }
-    
-    struct ErrorModel {
-        let message: String
-        let retry: () -> AnyPublisher<EditUserProfileViewState, Never>
-        let cancel: () -> AnyPublisher<EditUserProfileViewState, Never>
-    }
+}
+
+protocol EditingModeling {
+    func save(username: String) -> AnyPublisher<EditUserProfileViewState, Never>
+}
+
+protocol SavingErrorModeling {
+    let message: String
+    func retry() -> AnyPublisher<EditUserProfileViewState, Never>
+    func cancel() -> AnyPublisher<EditUserProfileViewState, Never>
 }
```

Comment:
? - I know a pattern some of us have used is to define a type alias for the state publisher type to avoid the verbosity of repeated `AnyPublisher` definitions.  Is this a convention that we want to explicitly recommend?  Or does that just add unnecessary complexity to these examples?

Reply:
Good callout! I think it's an unnecessary abstraction for this example because the "Use `Never` and return only view states" concept is still new to the reader, so I don't want to abstract from it.

Aside from that, it may not be a good pattern even in regular practice because it is an abstraction that removes the reader from the source of truth while saving only a handful of keystrokes (11 in this case):

`AnyPublisher<EditUserProfileViewState, Never>` vs `EditUserProfileViewState.Publisher`

I'm not saying that it shouldn't be done, but I feel that the case for it is not strong enough to sell in the documentation. Let's meet up to discuss this offline.